### ISSUE TITLE
Pipewire 0.3.28: Couldn't create pulseaudio client: No valid path for Pulse cookie found

### ISSUE THREAD

**Issue by kipcd** [lawl/NoiseTorch#136]
Hi,
I've just installed the noisetorch 0.11.2 from AUR. When I'm trying to run NoiseTorch it outputs an error:
```
$ noisetorch
Couldn't create pulseaudio client: No valid path for Pulse cookie found
```
Distro: Arch Linux
extra/pipewire-pulse 1:0.3.28-1 (Installed)
extra/pipewire-media-session 1:0.3.28-1 (Installed)
extra/pipewire-alsa 1:0.3.28-1 (Installed)
extra/pipewire 1:0.3.28-1 (Installed)

Any way to debug it? Thanks!

[noisetorch.log](https://github.com/lawl/NoiseTorch/files/6523973/noisetorch.log)

**Comment by lawl** [comment]
Can you attach a log? See: https://github.com/lawl/NoiseTorch/wiki/Troubleshooting

**Comment by lawl** [comment]
Generally, the code to resolve the cookie path is here: https://github.com/lawl/pulseaudio/blob/ab0735955fd096c04b6ceda41350b6febb15090a/client.go#L367

So if it's not in a default location, `PULSE_COOKIE_PATH` should be set. Do other pulseaudio tools like `pactl` work fine for you? (Yes, pulseaudio, we still use the pulseaudio bridge on pipewire).

I'm wondering because this code to resolve the cookie seems to have worked fine for a long time now.

**Comment by kipcd** [comment]
Found it! 
I have a cookie in `~/.pulse-cookie`, with hyphen instead of underscore. I don't know why, never touch it nor configure anything related. Could be the default name? 
pactl is working:
```
❯ pactl info
Server String: /run/user/1000/pulse/native
Library Protocol Version: 34
Server Protocol Version: 35
Is Local: yes
Client Index: 56
Tile Size: 65472
User Name: denis
Host Name: archmsi
Server Name: PulseAudio (on PipeWire 0.3.28)
Server Version: 14.0.0
Default Sample Specification: float32le 2ch 48000Hz
Default Channel Map: front-left,front-right
Default Sink: alsa_output.pci-0000_00_1f.3.analog-stereo
Default Source: alsa_input.pci-0000_00_1f.3.analog-stereo
Cookie: 927b:1c6d
```

**Comment by lawl** [comment]
https://github.com/pulseaudio/pulseaudio/blob/7f4d7fcf5f6407913e50604c6195d0d5356195b1/src/pulsecore/native-common.h#L195

Whoops, it's also supposed to be a hyphen. Thanks for finding this!

In the meantime you should be able to work around this bug by starting noisetorch with

`PULSE_COOKIE=~/.pulse-cookie noisetorch`

**Comment by kipcd** [comment]
Yes! It works with a cookie. Will test it on the next daily meeting :)
Thank you very much!

**Comment by eskoONE** [comment]
i added `PULSE_COOKIE=~/.pulse-cookie noisetorch` to .config/noisetorch/config.toml but now im getting 
```$ sudo noisetorch
Couldn't create pulseaudio client: No valid directory for Pulse RuntimePath found

do i have to do anything else?

**Comment by wosym** [comment]
Same issue here. Also Arch user. (I installed through AUR)
I have the `Couldn't create pulseaudio client: No valid path for Pulse cookie found` issue, but this is indeed resolved by `PULSE_COOKIE=~/.pulse-cookie noisetorch`. 
The GUI loads up and asks for root privileges. However, when I enter the password it says "Error: exit status 127". 
When I try to start it with sudo from the start, I get the same error as above:
`Couldn't create pulseaudio client: No valid directory for Pulse RuntimePath found`

**Comment by lawl** [comment]
This is fixed in [0.11.3](https://github.com/lawl/NoiseTorch/releases/tag/0.11.3)

@wosym Only saw the last part now. Can you open a new issue if the RuntimePath issue persists?
